scripts/screeps_rl_dashboard.py

```
@@ -309,6 +387,199 @@ def first_present(value: JsonObject, keys: Sequence[str]) -> Any:
+def preflight_marker_present(payload: JsonObject) -> bool:
+    return any(
+        controller_summary_final_status_key(node) in PREFLIGHT_FINAL_STATUS_KEYS
+        for node in iter_json_objects(payload)
+    )
+
+
+def collect_strong_compute_evidence(payload: JsonObject) -> list[JsonObject]:
+    signals: list[JsonObject] = []
+    seen: set[tuple[str, str]] = set()
+
+    def add(field: str, value: Any) -> None:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)